AVanyan/assignment-3/Lexicon.java

```
@@ -0,0 +1,67 @@
+import org.mockito.internal.matchers.LessOrEqual;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.File;
+import java.util.Scanner;
+
+/**
+ * This is an implementation of lexicon, it contains to methods: isWord which takes string and returns whether string
+ * is a word in the lexicon, isPrefix which takes string and returns whether the string is a prefix of at least one
+ * word in the lexicon.
+ */
+public class Lexicon {
+    private List <String> lexiconWords;
+
+    Lexicon(String filename) {
+
+        Scanner scanner = null;
+        try {
+            scanner = new Scanner(new File(filename));
+            lexiconWords = new ArrayList<>();
+            while (scanner.hasNext()) {
+                lexiconWords.add(scanner.next());
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
```

It is better to throw an exception, than just to print it and continue, because if there was an exception in the try block, it signals that something went wrong, and lexiconWords can be invalid.